R/sprintf_linter.R

```diff
@@ -59,57 +118,3 @@ sprintf_linter <- function() {
     )
   })
 }
-
-#' Zap sprintf() call to contain only constants
-#'
-#' Set all extra arguments to 0L if they aren't a constant
-#'
-#' @param parsed_expr A parsed `sprintf()` call
-#'
-#' @return A `sprintf()` call with all non-constants replaced by `0L`
-#' (which is compatible with all sprintf format specifiers)
-#'
-#' @noRd
-zap_extra_args <- function(parsed_expr) {
-  is_missing <- function(x) {
-    is.symbol(x) && !nzchar(x)
-  }
-
-  if ("fmt" %in% names(parsed_expr)) {
-    fmt_loc <- which(names(parsed_expr) == "fmt")
-  } else {
-    fmt_loc <- 2L
-  }
-
-  if (length(parsed_expr) >= 3L) {
-    for (i in setdiff(seq_along(parsed_expr), c(1L, fmt_loc))) {
-      if (!is_missing(parsed_expr[[i]]) && !is.atomic(parsed_expr[[i]])) {
-        parsed_expr[[i]] <- 0L
-      }
-    }
-  }
-  parsed_expr
-}
-
-#' Anticipate warnings of a sprintf() call
-#'
-#' Try running a static sprintf() call to determine whether it will produce warnings or errors due to format
-#' misspecification
-#'
-#' @param xml An XML node representing a `sprintf()` call (i.e. the `<expr>` node containing the call)
-#'
-#' @return A string, either `NA_character_` or the text of generated errors and warnings from the `sprintf()` call when
-#' replacing all dynamic components by 0, which is compatible with all format specifiers.
-#'
-#' @noRd
-capture_sprintf_warning <- function(xml) {
```

Comment:
Moving the functions inside cleans up the namespace, but why remove the roxygen markup?

Reply:
it will get included in the parent function's roxygen, iinm. that's how R6 documentation works... let me confirm

Reply:
Yep, if I restored the `#'` above `zap_extra_args()`, we get this roxygen2 warning from `parse_file("R/sprintf_linter.R")`:

```
Warning message:
[sprintf_linter.R:28] @export must be a single line, not 3
ℹ The first line is "Zap sprintf() call to contain only constants" 
```

because the `#' @export` line terminating the `sprintf_linter` assignment is being extended to include the beginning of the internal function's header.